internal/docker_test.go

```diff
@@ -0,0 +1,515 @@
+func (m *mockDockerClient) ContainerList(ctx context.Context,
+	options types.ContainerListOptions) ([]types.Container, error) {
+	m.ContainerListInvoked++
+
+	return m.ContainerListFn()
+}
+
+// ContainerKill is a mock function which kills (removes entry of) a container
+// given its ID.
+func (m *mockDockerClient) ContainerKill(ctx context.Context, containerID, signal string) error {
+	m.ContainerKillInvoked++
+
+	return m.ContainerKillFn(containerID)
+}
+
+// ContainerStats is a mock function which returns the mem and cpu stats of a container
+// given its ID.
+func (m *mockDockerClient) ContainerStats(ctx context.Context,
+	containerID string, stream bool) (types.ContainerStats, error) {
+	m.ContainerStatsInvoked++
+
+	return m.ContainerStatsFn(containerID)
+}
+
+const (
+	readerCloserStats = `{
+		"read":"2021-01-05T11:42:54.959351591Z",
+		"preread":"2021-01-05T11:42:53.949728039Z",
+		"pids_stats":{"current":4},
+		"blkio_stats":{},
+		"num_procs":0,
+		"storage_stats":{},
+		"cpu_stats":{
+			"cpu_usage":{
+				"total_usage":1244741231366,
+				"percpu_usage":[924236203020,320505028346],
+				"usage_in_kernelmode":9190000000,
+				"usage_in_usermode":653150000000
+			},
+			"system_cpu_usage":2053540000000,
+			"online_cpus":2,
+			"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}
+		},
+		"precpu_stats":{},
+		"memory_stats":{
+			"usage":57921536,
+			"max_usage":115904512,
+			"stats":{
+				"active_anon":1216512,
+				"active_file":41766912,
+				"cache":53268480,
+				"dirty":135168,
+				"hierarchical_memory_limit":9223372036854771712,
+				"hierarchical_memsw_limit":9223372036854771712,
+				"inactive_anon":0,
+				"inactive_file":11354112,
+				"mapped_file":3514368,
+				"pgfault":97911,
+				"pgmajfault":165,
+				"pgpgin":66198,
+				"pgpgout":52876,
+			    "rss":1048576,
+				"rss_huge":0,
+				"total_active_anon":1216512,
+				"total_active_file":41766912,
+				"total_cache":53268480,
+				"total_dirty":135168,
+				"total_inactive_anon":0,
+				"total_inactive_file":11354112,
+				"total_mapped_file":3514368,
+				"total_pgfault":97911,
+				"total_pgmajfault":165,
+				"total_pgpgin":66198,
+				"total_pgpgout":52876,
+				"total_rss":1048576,
+				"total_rss_huge":0,
+				"total_unevictable":0,
+				"total_writeback":0,
+				"unevictable":0,
+				"writeback":0},
+				"limit":2084458496
+			},
+			"name":"/test_container2",
+			"id":"container_id2",
+			"networks":{}
+		}`
+
+	dockerFileMode os.FileMode = 0600
+)
+
+// removeContainer removes the container from container list if found
+// and returns the remaining containers.
+func removeContainer(cntrList []types.Container, containerID string) []types.Container {
+	remainingCntr := []types.Container{}
+
+	for _, cntr := range cntrList {
+		if cntr.ID != containerID {
+			remainingCntr = append(remainingCntr, cntr)
+		}
+	}
+
+	return remainingCntr
+}
+
+// getContainerStats creates and returns the container's stats
```

Comment:
This should be more explicit that it is returning example test stats, not stats specific to the given containerID

Reply:
Updated